### DISCUSSION TITLE
useQuery data is undefined when refetching

### DISCUSSION THREAD

**Discussion by tomashornak** [TanStack/query#882] [General] [answered]
Hi, sorry if this is obvious but I find this behavior little bit annoying.

When useQuery is triggered by new input data, for time it is refetched, `data` variable becomes `undefined`. This makes react to rerender with empty data and then again render with new data. Do I have to cache response data manually or is there some way how to stop this behavior without seeing blank screen between each refetch. I was not able to find built-in solution.

Please let me know if you need sample code.
Thank you

**Comment by boschni** [comment, marked as answer]
Hi @tomashornak! Do you mean `data` becomes `undefined` when the query key changes? Because this should be addressed in this PR: https://github.com/tannerlinsley/react-query/pull/867

**Reply by tomashornak** [reply]
Thank you for answer.
Yes this looks what I'm looking for.
So for now solution would be to use `usePaginatedQuery` with `resolvedData`.

**Reply by oybekalimat** [reply]
Or add `keepPreviousData` option
